### PR TITLE
Add method for getting transaction level advisory locks in Postgresql

### PR DESCRIPTION
**Pull request by thgreasi**
Change-type: minor
Signed-off-by: Thodoris Greasidis <thodoris@balena.io>

### CODE REVIEW

File: src/database-layer/db.ts [hunk 1]
```diff
@@ -529,6 +535,20 @@ if (maybePg != null) {
 				}
 			}
 
+			public override async getTxLevelLock(key1: number, key2: number) {
```
Comment (Page-) [inline src/database-layer/db.ts:538]: I'd prefer to see the first key be pre-registered, eg how open-balena-api uses `ADVISORY_LOCK_NAMESPACES`, you'd pre-register your namespaces and then just provide the namespace key for the first argument

Comment (thgreasi) [reply src/database-layer/db.ts:538]: Should that be done by a `registerLockNamespace` method exported by `database-layer/db` or would you imagine it be better if it's added as an extra property in `init`'s `config` object (in `server-glue/module`).
At first a `registerLockNamespace()` that each feature can register it's own namespaces sounded like more appropriate for the feature based separation that we have started following, but on second thought, we will still need coordination in order for two features to not end up with a namespace id clash.

File: src/database-layer/db.ts [hunk 2]
```diff
@@ -108,6 +108,32 @@ export const engines: {
 	[engine: string]: (connectString: string | object) => Database;
 } = {};
 
+const transactionLockNamespaceMap: Dictionary<number> = {};
+
+export function registerTransactionLockNamespace(
+	namespaceKey: string,
+	namespaceId: number,
```
Comment (Page-) [inline src/database-layer/db.ts:115]: This is an integer type in postgres so at least for that needs to be within -2147483648 to +2147483647 (https://www.postgresql.org/docs/13/datatype-numeric.html), I think for now just checking 0-2147483647 would be fine - we can always expand it later as need be if we for some reason want more

File: src/database-layer/db.ts [hunk 3]
```diff
@@ -108,6 +108,32 @@ export const engines: {
...
+		throw new TypeError(`Invalid namespaceId parameter provided for Tx lock`);
+	}
+	if (transactionLockNamespaceMap[namespaceKey] != null) {
+		throw new Error(
+			`Error while registering transaction lock namespace '${namespaceKey}'. Namespace key is already registered.`,
+		);
+	}
+	const existingNamespaceEntry = Object.entries(
+		transactionLockNamespaceMap,
+	).find(([_key, id]) => id === namespaceId);
```
Comment (Page-) [inline src/database-layer/db.ts:127]:
```suggestion
	).find(([, id]) => id === namespaceId);
```

**Review by Page- — APPROVED** [review]

**Comment by thgreasi** [comment]
@balena-ci rebase

**Comment by thgreasi** [comment]
@balena-ci rebase

**Pull request merged**